src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -863,9 +863,10 @@ public void EnsureNotDeleted(string node)
 
         public void CompleteDeletionIfNeeded(CancellationTokenSource cts)
         {
+            var dbName = Database.Name;
```

Comment:
Do we need to check for null here?

Reply:
The race is that we execute on the `ThreadPool` after the dispose happened where we set the `Database` to null.
We don't need to check for `null` here because we wait for this thread to end before setting to `Database` to null.